pkg/machinery/client/client.go

```diff
@@ -360,6 +360,11 @@ func (c *Client) Kubeconfig(ctx context.Context) ([]byte, error) {
 	return kubeconfig, err
 }
 
+// ApplyConfiguration implements proto.OSClient interface.
+func (c *Client) ApplyConfiguration(ctx context.Context, req *machineapi.ApplyConfigurationRequest, callOptions ...grpc.CallOption) (resp *machineapi.ApplyConfigurationResponse, err error) {
```

Comment:
All of these methods are labeled as "implements proto.OSCLient", but that is not actually true, so far as I can tell.  As such, it could be easily argued that the simpler form of accepting just the data and returning just the error is acceptable here.  However, since the options may grow over time, I decided to just include the canonical form.  I am not married to this.

Reply:
on other hand, to get "raw" API one can always just use `c.MachineClient.ApplyConfiguration` directly. I don't know what's the best way here, I had same thoughts as well